api/desecapi/pdns.py

```diff
@@ -121,6 +126,40 @@ def delete_zone(name):
     _pdns_delete('/zones/' + normalize_hostname(name))
 
 
+def get_zone(name):
+    """
+    Retrieves a JSON representation of the zone from pdns
+    """
+    r = _pdns_get('/zones/' + normalize_hostname(name))
+
+    return r.json()
+
+
+def get_rrsets(name, subname = None, type = None):
+    """
+    Retrieves a JSON representation of the RRsets in a given zone, optionally restricting to a name and RRset type 
+    """
+    fullname = '.'.join(filter(None, [subname, name])) + '.'
+
+    rrsets = get_zone(name)['rrsets']
+    rrsets = [rrset for rrset in rrsets \
+              if (subname == None or rrset['name'] == fullname) and (type == None or rrset['type'] == type) \
+        ]
+
+    # Make sure that if subname and type are given, we get one or none RRsets
+    assert subname == None or type == None or len(rrsets) <= 1, \
+            "pdns returned multiple RRsets of type %s for %s (zone: %s)" % (type, fullname, name)
+
+    return jq("[.[] | {name: .name, records: [.records[] | .content], ttl: .ttl, type: .type}]").transform(rrsets)
```

Comment:
This interface breaks current `pdns.py` design pattern: up to now, all `pdns.py` interfaces are ("proper") python objects; `get_rrsets` and `set_rrsets` however both rely on serialized JSON objects. Unless there is good reason, this should be changed to python objects.